### ISSUE TITLE
Set the image height attribute, but it does not work

### ISSUE THREAD

**Issue by diguage** [asciidoctor/asciidoctor#2542]
I set the image height atrrribute as following:

```
image:my.jpg[My Image,width=640,height=480]
```

It output HTML as following:

```
<img alt="My Image" src="my.jpg" width="640" height="480">
```

But it does not work.

In my opinion, it output HTML as following:

```
<img alt="My Image" src="my.jpg" style="width:640;height:480;">
```

It's OK.

**Comment by mojavelinux** [comment]
The problem is, that breaks responsiveness of the image (when browser window resizes, it won't scale proportionally). Responsive images and setting both dimensions is fundamentally incompatible. This is where I advise having a custom template if you really want that behavior.

There's another issue where we discussed this if you are interested in a deeper discussion. See https://github.com/asciidoctor/asciidoctor/issues/1116#issuecomment-102783590

**Comment by cirosantilli** [comment]
Is there any way to force adding `style="width:640;height:480;"`? I could not find a valid syntax.

The only workaround I can find is with a `[.myclass]` + `docinfo.html` CSS, but I can't find the syntax to add a class to an inline image `image:`.

Maybe there is good rationale for current behavior, but the current interface is just too unanatomic: when an user writes width= height=, then it feels really bad when this does not happen and you can't find a way after 15 minutes search :-) Better let those users shoot themselves in the foot if they want.

adoc 2.0.10.

**Comment by mojavelinux** [comment]
It's not about letting users shoot themselves in the foot. Setting a width and height explicitly, and having responsive behavior on the web are fundamentally incompatible. We didn't make that decision. It's how browsers work. So the only reasonable way to be consistent with responsive browser behavior is to ignore the height (and maintain the aspect ratio).

If you want to override the default stylesheet, that is well within your capacity to do so. But keep in mind, when you allow height to be set, it breaks responsive behavior.

**Comment by mojavelinux** [comment]
I strongly recommend joining us in the gitter channel as questions like this would be better answered there. https://gitter.im/asciidoctor/asciidoctor

**Comment by cirosantilli** [comment]
Thanks Dan,

I'm OK with breaking responsive on my document, I just want that height to be fixed :-)

Is there a good way to override just this parameter on the default stylesheet? I like the rest of it, just want to change this. Or a way to set arbitrary inline style= on images?

If that is possible, it would be good to document this in a comment of one of the GitHub issues for other users who also want to break responsive :-)

**Comment by mojavelinux** [comment]
You're going to find yourself going against the grain here because it's not Asciidoctor (or the default stylesheet) that's making this difficult to achieve, it's the browser.

For example, you might assume you could use:

```
img {
  height: unset;
}
```

Trust me, I tried that. But the browser just ignores it (since the height has previously been set to auto in the stylesheet).

The browser works very hard to prevent you from modifying the aspect ratio of an image. That's just how it is. If someone wants to show me how this can work (using overrides), I'm happy to learn. But I just don't see a way.

> Or a way to set arbitrary inline style= on images

That would completely break the AsciiDoc contract, so no. You must use roles.

One thing we could do is add support for disabling the image responsiveness using an option like `noresponsive`.

```
image::tiger.png[tiger,200,400,opts=noresponsive]
```

That would add the `non-responsive` class to the img element and prevent the max-width and height from being set in the default stylesheet. (We might have to work on the terminology).

**Comment by mojavelinux** [comment]
Btw, what's the use case for messing with the aspect ratio of the image. It's going to look distorted. Is that what you want? Is there I reason you can't set the image size correctly in the original image?

**Comment by cirosantilli** [comment]
Hmmm, I only want to set a fixed height, not both height and width at the same time. Could that distort the image?

Use case: these are going to be the very first introductoin images on my homepage, so they need to be small enough in height so that the ToC is easily visible below.

I might end up going for `-a stylesheet!` or inline HTML `<img>` elements :-)

**Comment by cirosantilli** [comment]
The non-responsive would be a good workaround however :+1: I'm happy with anything that does the trick.

**Comment by mojavelinux** [comment]
Why don't you just figure out which width gives you the height you want and set that? I still stand by the fact that what you're asking for goes against how a browser works. Browsers are biased towards the width dimension. So you really need to set the width and have the height be implied. We have to work with the medium that we have.

**Comment by mojavelinux** [comment]
(I'm still open to the idea of disabling the setting via an explicit option as it seems like a fair compromise for how to interpret the width and height attributes).

**Comment by Mogztter** [comment]
I think @cirosantilli use case is to define a maximum height.
The image should not take more than X in vertical space otherwise other important elements won't be seen.

**Comment by cirosantilli** [comment]
@Mogztter yes, that would work too. Is there a way to set it inline so that it "beats" the CSS header?

I'm already setting the height to what feels like a minimal reasonable height though, so I guess if a screen is not large enough even for that, the end users just needs a larger screen :-)

**Comment by mojavelinux** [comment]
> The image should not take more than X in vertical space otherwise other important elements won't be seen.

You could wrap the image in an open block and give it a style that limits its height.

It's just not possible to do this directly on the image, even if you used hardcoded CSS because the browser just doesn't support that use case. Trust me when I say I've tried.
